### PR TITLE
daemon,client,cmd/snap: nice access denied message

### PR DESCRIPTION
**Pull request by niemeyer**
Based on both #1136 and mvo's #1091.

Changes on top of the previously approved mvo branch may be seen here: https://github.com/ubuntu-core/snappy/pull/1137/commits/6bdd8e9

Closes: LP#1574829 #

### CODE REVIEW

**Comment by pedronis** [comment]
LGTM

File: cmd/snap/main.go [hunk 1]
```diff
@@ -151,8 +149,9 @@ func run() error {
 			return nil
 
 		}
-		if e, ok := err.(*client.Error); ok && e.StatusCode == 401 && e.Kind == client.ErrorKindLoginRequired {
-			return fmt.Errorf(missingLoginHelp)
+		if e, ok := err.(*client.Error); e.Kind == client.ErrorKindLoginRequired {
```
Comment (chipaca) [inline cmd/snap/main.go]: hmm, this seems wrong. Either the `ok` should still be checked, or not assigned?

Comment (chipaca) [inline cmd/snap/main.go]: replying to myself: fixed in 84f4cb3262cb33718c9dd43f495d41d4a048377a

**Comment by chipaca** [comment]
LGTM. A bit of a cow of a branch, and I'd like to take a secodn look once #1136 is merged and this rebased/merged, but LGTM.

**Comment by chipaca** [comment]
:+1:

**Pull request merged**